ElectronClient/app.js

```diff
@@ -51,6 +51,7 @@ const appDefaultState = Object.assign({}, defaultState, {
 	watchedNoteFiles: [],
 	lastEditorScrollPercents: {},
 	devToolsVisible: false,
+	isNightMode: false,
```

Comment:
I didn't check all the code yet, but please see if you can avoid the need to create a new state variable. I expect Setting should have all the data you need, and you can use `SETTING_UPDATE_ONE` to check when particular settings are being changed.

Reply:
the state variable is needed to toggle Night Mode button ON/OFF just like Visible Panes setting, I have also added `isNightMode` key in settings to save last button state so it can be retrieved on restarting the app

Reply:
Night Mode button is similar to Toggle note list button, only difference is Toggle note list button changes Notes Panel visibility and Night Mode button changes theme, So I simply followed the Toggle note list button implementation method for Night Mode button.